crates/forge/src/collecting.rs

```diff
@@ -0,0 +1,588 @@
+                    name: "crate1::do_thing".to_string(),
+                    available_gas: None,
+                    expected_result: ExpectedTestResult::Success,
+                    fork_config: None,
+                    fuzzer_config: None,
+                },
+                TestCase {
+                    name: "crate2::run_other_thing".to_string(),
+                    available_gas: None,
+                    expected_result: ExpectedTestResult::Success,
+                    fork_config: None,
+                    fuzzer_config: None,
+                },
+                TestCase {
+                    name: "outer::crate3::run_other_thing".to_string(),
+                    available_gas: None,
+                    expected_result: ExpectedTestResult::Success,
+                    fork_config: None,
+                    fuzzer_config: None,
+                },
+                TestCase {
+                    name: "do_thing".to_string(),
+                    available_gas: None,
+                    expected_result: ExpectedTestResult::Success,
+                    fork_config: None,
+                    fuzzer_config: None,
+                },
+            ],
+            test_crate_type: TestCrateType::Tests,
+        };
+
+        let filtered = mocked_tests.filter_by_name("", true);
+        assert_eq!(filtered.test_cases, vec![]);
+
+        let filtered = mocked_tests.filter_by_name("thing", true);
+        assert_eq!(filtered.test_cases, vec![]);
+
+        let filtered = mocked_tests.filter_by_name("do_thing", true);
+        assert_eq!(
+            filtered.test_cases,
+            vec![TestCase {
+                name: "do_thing".to_string(),
+                available_gas: None,
+                expected_result: ExpectedTestResult::Success,
+                fork_config: None,
+                fuzzer_config: None,
+            },]
+        );
+
+        let filtered = mocked_tests.filter_by_name("crate1::do_thing", true);
+        assert_eq!(
+            filtered.test_cases,
+            vec![TestCase {
+                name: "crate1::do_thing".to_string(),
+                available_gas: None,
+                expected_result: ExpectedTestResult::Success,
+                fork_config: None,
+                fuzzer_config: None,
+            },]
+        );
+
+        let filtered = mocked_tests.filter_by_name("crate3::run_other_thing", true);
+        assert_eq!(filtered.test_cases, vec![]);
+
+        let filtered = mocked_tests.filter_by_name("outer::crate3::run_other_thing", true);
+        assert_eq!(
+            filtered.test_cases,
+            vec![TestCase {
+                name: "outer::crate3::run_other_thing".to_string(),
+                available_gas: None,
+                expected_result: ExpectedTestResult::Success,
+                fork_config: None,
+                fuzzer_config: None,
+            },]
+        );
+    }
+
+    #[test]
+    fn filtering_tests_works_without_crate_in_test_name() {
```

Comment:
I don't understand what this test verifies

Reply:
It was meant to verify that even for tests which somehow doesn't have a crate name attached (no part before `::`) the filtering logic works correctly. I'll combine it with other tests though.

Reply:
I am not sure it this test actually test something useful. It is kinda testing logic of `name.contains` which is part of the std lib

Reply:
Yeah, it's been removed, I've just extanded mocked_tests in other unit tests